File: dbaird1_hw1/ack/ack.rb
#! /usr/bin/env ruby
def ack(m,n)
  puts n
  if(m == 0)
      return n+1;
  end
  if(n == 0 && m > 0)
      return ack(m-1, 1);
  end
  if(n>0 && m > 0)
      return ack(m-1,ack(m, n-1));
  end
  return 0;
end



if ARGV.length != 1
  puts "fact.rb usage: [NUMBER]"
  exit
end

ack(3,ARGV[0].to_i);

File: dbaird1_hw1/ack/ack.py
#! /usr/bin/env python

import sys

def ack(m,n):
    print(n)
    if(m == 0):
        return n+1;
    if(n == 0 and m > 0):
        return ack(m-1, 1);
    if(n>0 and m > 0):
        return ack(m-1,ack(m, n-1));
    return 1;



if len(sys.argv) != 2:
  print("%s usage: [NUMBER]" % sys.argv[0])
  exit()

ack(3,int(sys.argv[1]))
